File: templates/griffon-javafx-java-templates/templates/griffon-javafx-java/src/main/java/ApplicationEventHandler.java
package ${project_package};

import griffon.core.event.EventHandler;
import griffon.exceptions.GriffonViewInitializationException;
import javafx.application.Platform;

public class ApplicationEventHandler implements EventHandler {
    public void onUncaughtGriffonViewInitializationException(GriffonViewInitializationException x) {
        Platform.exit();
    }
}

File: templates/griffon-javafx-kotlin-templates/templates/griffon-javafx-kotlin/src/main/kotlin/ApplicationEventHandler.kt
package ${project_package}

import griffon.core.event.EventHandler
import griffon.exceptions.GriffonViewInitializationException
import javafx.application.Platform

class ApplicationEventHandler : EventHandler {
    @javax.application.event.EventHandler
    fun handleUncaughtExceptionThrownEvent(x: UncaughtExceptionThrownEvent) {
        if (x.getThrowable() is GriffonViewInitializationException) {
            Platform.exit()
        }
    }
}